hubtools/src/lib.rs

```diff
@@ -602,4 +662,125 @@ impl RawHubrisArchive {
             .copy_from_slice(input.as_bytes());
         Ok(())
     }
+
+    /// Verifies the signature of an LPC55 image
+    ///
+    /// Results are printed to `stderr` using `log`
+    pub fn verify(&self, verbose: bool) -> Result<(), Error> {
+        // CMPA and CFPA are included in the archive (for now)
+        let cmpa_bytes = self.extract_file("img/CMPA.bin")?;
+        if cmpa_bytes.len() != 512 {
+            return Err(Error::BadCMPASize(cmpa_bytes.len()));
+        }
+        let mut cmpa_array = [0u8; 512];
```

Comment:
I've been trying to reduce use of this pattern in our tools -- we do this because packed_struct wants to take a fixed-length array but things like reading from a file don't produce a fixed-length static array. We don't need to copy to achieve this.

You can perform  the length check (above) and statically encode the result in the type by converting the vec to a `Box<[u8; 512]>` using `TryFrom`. This does not involve a copy -- if the vec is the right size, it reuses the allocation, possibly shrinking it if there was excess capacity.

```rust
let cmpa_bytes: Box<[u8; 512]> = cmpa_bytes.try_into()
    .map_err(|v| Error::BadCMPASize(v.len()))?;
    
let cmpa = lpc55_areas::CMPAPage::from_bytes(&cmpa_bytes)?;
```

Reply:
Neat, done!